### PR TITLE
Update Jupyter Notebook Upload Related

### PR DESCRIPTION
**Pull request by mengting1010**
Description of changeset:
- Integrate with S3 client 
- upload Jupyter Notebook to s3 when saving the post 
- export a html version of Jupyter Notebook and upload to s3

Test Plan:
local dev

Reviewers:
@csharplus @JJJ000

### CODE REVIEW

**Review by csharplus — APPROVED** [review]
LGTM.

File: knowledge_repo/app/routes/editor.py [hunk 1]
```diff
@@ -195,8 +195,8 @@ def save_post():
     if "ipynb" in data:
         headers["ipynb"] = data["ipynb"]
         if (
-            data.get("file_name", None) is not None
-            and data.get("file_data", None) is not None
+            data.get("file_name", None) is not None and \
```
Comment (csharplus) [inline knowledge_repo/app/routes/editor.py:198]: You got Lint error in CI:
```
knowledge_repo/app/routes/editor.py:198:57: E502 the backslash is redundant between brackets
Error: Process completed with exit code 1.
```

File: knowledge_repo/app/routes/editor.py [hunk 2]
```diff
@@ -128,8 +128,8 @@ def editor(path=None):
             )
 
     if (
-        current_user.identifier not in data["authors"]
-        or current_user.identifier in current_repo.config.editors
+        current_user.identifier not in data["authors"] or \
```
Comment (csharplus) [inline knowledge_repo/app/routes/editor.py:131]: Same lint issue here.

File: knowledge_repo/app/routes/editor.py [hunk 3]
```diff
@@ -168,8 +168,8 @@ def save_post():
     if path in current_repo:
         kp = current_repo.post(path)
         if (
-            current_user.identifier not in kp.headers["authors"]
-            and current_user.identifier not in current_repo.config.editors
+            current_user.identifier not in kp.headers["authors"] and \
```
Comment (csharplus) [inline knowledge_repo/app/routes/editor.py:171]: Same lint issue here.

**Review by JJJ000 — APPROVED** [review]

**Pull request merged**